pkg/virt-controller/services/template.go

```diff
@@ -425,6 +427,18 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, i
 		compute.Env = append(compute.Env, k8sv1.EnvVar{Name: ENV_VAR_VIRTIOFSD_DEBUG_LOGS, Value: "1"})
 	}
 
+	if vmi.Spec.Domain.Devices.DownwardMetrics != nil {
```

Comment:
Setting the env vars unconditionally without checking for the DownwardMetrics device, will make the virt-launcher to fail because qemu will not create the virt-serial socket